tools/windows-node-installer/pkg/client/client.go

```diff
@@ -3,26 +3,61 @@ package client
 import (
 	"fmt"
 
-	client "github.com/openshift/client-go/config/clientset/versioned"
-	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/openshift"
+	v1 "github.com/openshift/api/config/v1"
+	clientset "github.com/openshift/client-go/config/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/tools/clientcmd"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	logger "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
-// log is the global logger for the client package. logf uses logr generic interface. Each log record produced
-// by this logger will have an identifier containing `client` tag.
-var log = logf.Log.WithName("client")
+// log is the global logger for the client package.
+// Each log record produced by this logger will have an identifier containing `client` tag.
+var log = logger.Log.WithName("client")
 
-// GetOpenShiftClient uses kubeconfig to create a client for existing OpenShift cluster and returns it or an error.
-func GetOpenShiftClient(kubeConfigPath string) (*openshift.Client, error) {
+// Client is an openShiftClient struct which will be used for all OpenShift related functions to interact with
+// the existing Cluster.
+type Client struct {
```

Comment:
I think this OpenShift `Client` should be exposed because the functions belong to this struct is used in `ec2.go`. If we rely on GetOpenShift in every package, we would need to pass kubeconfig path. We will be expanding functionalities here including Getting infraID which was moved to the next PR.

Reply:
I don't understand why `Client` should be exported. You instantiate the client using `GetOpenShift()` and reuse it.